File: example/typescript/main/SomeModule/Impl/Logic.ts
namespace SomeModule.Impl {
    export class SomeEmptyInterfaceLogic implements SomeEmptyInterface {
        constructor(
            private readonly c: HandlerContext,
        ) {}
    }

    export class SomeInterfaceLogic implements SomeInterface {
        constructor(
            private readonly c: HandlerContext,
        ) {}

        someEmptyMethod(): void {
            // TODO
            return undefined
        }

        someCommand(id: SomeId, amount: number): void {
            // TODO
            return undefined
        }

        someQuery(query: SomeQueryInput): SomeClass {
            // TODO
            return undefined
        }

        optMethod(optId: Optional<SomeId>): Optional<SomeClass> {
            // TODO
            return undefined
        }
    }

    export class SomeInterface2Logic implements SomeInterface2 {
        constructor(
            private readonly c: HandlerContext,
        ) {}

        referenceOtherClass(other: OtherClass): OtherClass {
            // TODO
            return undefined
        }

        referenceLegacyType(legacyType: LegacyType): LegacyType {
            // TODO
            return undefined
        }
    }

    export class SomeInterface3Logic implements SomeInterface3 {
        constructor(
            private readonly c: HandlerContext,
        ) {}

        referenceInterface(empty: SomeEmptyInterface): SomeEmptyInterface {
            // TODO
            return undefined
        }

        referenceOtherInterface(other: OtherInterface): OtherInterface {
            // TODO
            return undefined
        }
    }

    export class SomeModuleHandlersLogic implements SomeModuleHandlers {
        constructor(
            private readonly c: HandlerContext,
        ) {}

        someHandler(i: SomeHandlerInput): SomeHandlerOutput {
            // TODO
            return undefined
        }

        someHandler2(i: SomeHandlerInput): SomeHandlerOutput {
            // TODO
            return undefined
        }
    }

    export class SomeModuleDebugHandlersLogic implements SomeModuleDebugHandlers {
        constructor(
            private readonly c: HandlerContext,
        ) {}

        someDebugHandler(i: SomeHandlerInput): SomeHandlerOutput {
            // TODO
            return undefined
        }

        someDebugHandler2(i: SomeHandlerInput): SomeHandlerOutput {
            // TODO
            return undefined
        }
    }
}

File: example/kotlin/src/main/kotlin/com/some/pkg/somemodule/impl/Logic.kt
package com.some.pkg.somemodule.impl

import com.some.pkg.somemodule.api.*

import com.some.pkg.othermodule.api.*
import com.some.pkg.simplemodule.api.*
import com.some.pkg.typesmodule.api.*

class SomeEmptyInterfaceLogic: SomeEmptyInterface {
}

class SomeInterfaceLogic: SomeInterface {
    override fun someEmptyMethod(): Unit {
        TODO("Not yet implemented")
    }

    override fun someCommand(id: SomeId, amount: Int): Unit {
        TODO("Not yet implemented")
    }

    override fun someQuery(query: SomeQueryInput): SomeClass {
        TODO("Not yet implemented")
    }

    override fun optMethod(optId: SomeId?): SomeClass? {
        TODO("Not yet implemented")
    }

    override fun methodWithSimpleVO(id: SomeId): Unit {
        TODO("Not yet implemented")
    }

    override fun methodWithListOfSimpleVO(list: List<SomeId>): List<SomeId> {
        TODO("Not yet implemented")
    }

    override fun methodWithAny(i: Any): Any {
        TODO("Not yet implemented")
    }

    override fun methodWithBaseType(i: String): String {
        TODO("Not yet implemented")
    }

    override fun methodReturningOptSimpleVo(): SomeId? {
        TODO("Not yet implemented")
    }

    override fun methodReturningNumericType(): Int {
        TODO("Not yet implemented")
    }
}

class SomeInterface2Logic: SomeInterface2 {
    override fun referenceOtherClass(other: OtherClass): OtherClass {
        TODO("Not yet implemented")
    }

    override fun referenceLegacyType(legacyType: com.some.pkg.legacy.LegacyType): com.some.pkg.legacy.LegacyType {
        TODO("Not yet implemented")
    }
}

class SomeInterface3Logic: SomeInterface3 {
    override fun referenceInterface(empty: SomeEmptyInterface): SomeEmptyInterface {
        TODO("Not yet implemented")
    }

    override fun referenceOtherInterface(other: OtherInterface): OtherInterface {
        TODO("Not yet implemented")
    }
}

class SomeModuleHandlersLogic: SomeModuleHandlers {
    override fun someHandler(i: SomeHandlerInput): SomeHandlerOutput {
        TODO("Not yet implemented")
    }

    override fun someHandler2(i: SomeHandlerInput): SomeHandlerOutput {
        TODO("Not yet implemented")
    }
}

class SomeModuleDebugHandlersLogic: SomeModuleDebugHandlers {
    override fun someDebugHandler(i: SomeHandlerInput): SomeHandlerOutput {
        TODO("Not yet implemented")
    }

    override fun someDebugHandler2(i: SomeHandlerInput): SomeHandlerOutput {
        TODO("Not yet implemented")
    }
}

class InterfaceForTrackingLogic: InterfaceForTracking {
    override fun getDimension(): tracking.impl.TrackingDimension {
        TODO("Not yet implemented")
    }
}